src/mscorlib/src/System/String.cs

```diff
@@ -2174,43 +2175,48 @@ public static int Compare(String strA, String strB, StringComparison comparisonT
         // if this is equal to value, or a value greater than 0 if this is greater than value.
         //
         [Pure]
-        public int CompareTo(Object value) {
-            if (value == null) {
+        public int CompareTo(Object value)
+        {
+            if (value == null)
+            {
                 return 1;
             }
 
-            if (!(value is String)) {
+            string other = value as string;
+
+            if (other == null)
+            {
                 throw new ArgumentException(Environment.GetResourceString("Arg_MustBeString"));
             }
-
-            return String.Compare(this,(String)value, StringComparison.CurrentCulture);
+            
```

Comment:
@jamesqo, aware of that hack and agree that it doesn't matter, but i saw that in other places of this PR you have removed the extra whitespaces but these added extras at two other places.